### ISSUE TITLE
A number of centcom consoles have no ID or admin check

### ISSUE THREAD

**Issue by ArcLumin** [HippieStationArchive/HippieStation13-1#3026]
Both the operation of the centcom ferry at centcom and the usage of the Bluespace Cannon require no checks for accessing those consoles of any kind, nor does it ask the admins if they can use it, so if anyone accesses centcom and reaches those places they can access both, blowing huge huge swathes of area with the BSC or bringing more people on board by operating the ferry.

**Comment by JohnGinnane** [comment]
What sort of consoles are there, like the names of them?

**Comment by ArcLumin** [comment]
/obj/machinery/computer/shuttle/ferry for the ferry and
/obj/machinery/artillerycontrol for the BSC

**Comment by JohnGinnane** [comment]
Cheers

This is what is: https://github.com/HippieStationCode/HippieStation13/blob/56a3fdddeb58b6e0dedb20514f4cc43bee628c58/code/modules/awaymissions/bluespaceartillery.dm#L2

What it should be (if we decide heads can use these consoles): https://github.com/HippieStationCode/HippieStation13/blob/56a3fdddeb58b6e0dedb20514f4cc43bee628c58/code/game/machinery/computer/communications.dm#L11

**Comment by ArcLumin** [comment]
Yeah, I know. Just, dunno if we really should, yeah? I think it's pretty neat that we can do this shit, and it gives incentive to get to centcom. The BSC has a fairly large cooldown on it preventing it from fucking up the whole station.

**Comment by JohnGinnane** [comment]
It looks as though we need to specifically write the authentication for each console (if it requires it) as seen here: https://github.com/HippieStationCode/HippieStation13/blob/56a3fdddeb58b6e0dedb20514f4cc43bee628c58/code/game/machinery/computer/communications.dm#L65

I think we should implement a proc "auth_user" on the /machinery/ so it does that, so if you added a new console all you need to do is specify the req_access list (like it is in the communications console)

**Comment by JohnGinnane** [comment]
As a side note, we could make it so nobody has access to it, but the point is there's no authentication as you pointed out. Just using the comms console as a comparison

**Comment by ArcLumin** [comment]
Wouldn't be much point if nobody had access to it. They're there for players to use, not admins, as admins can just use commands for it. It's likely not so much of a bug as an oversight. Still, I think there should be some admin pings when it happens, not so much about blocking access. It's fine if it's all access as centcom's not easily accessible, but it should at least ping admins that "someone is in centcom and using the machinery".

**Comment by ghost** [comment]
At the very least, it should ping admins. Although BSA requires +FUN (I think). I think an admin ping in logs with various JUMP TO, TP, PP, ?, PM things should be sufficient.

**Comment by JohnGinnane** [comment]
That makes sense. So the usual message_admins()?

**Comment by ghost** [comment]
If message_admins() has those things then yeah sure.

**Comment by ghost** [comment]
If you could add an href to jump to the player as they do it, that would be optimal.

**Comment by JohnGinnane** [comment]
Yeah I'm thinking of using the code for admin PMs/prayers that has the <a href>TP</a> in the message just for that

**Comment by Swagile** [comment]
Yeah, if a player actually spends half their round trying to get to Centcomm, they should be rewarded for it considering they could be doing some other shit like blowing up the station using toxins (less time ), RnD (still less time), or just being a murderboning scrub with ebow + esword which would do infinitely more damage than a Centcomm BSA or allowing people into Centcomm via ferry.

**Comment by ArcLumin** [comment]
You CAN access like, the centcom armory as well, but you can also access the nuke ops shuttle and the blastco room so hey

**Comment by TDcoolguy300** [comment]
How about we have it so it just gibs anyone in Z2 when they are not suppose to be there

**Comment by ArcLumin** [comment]
How would it check? Shuttles naturally go to Z2, wizard stuff goes to Z2, abductors and abductees go to Z2, nuke ops are in Z2, arriving at centcom is at Z2, centcom recovery shit is at Z2, thunderdome, holodeck areas, admin funrooms, ferry, all that is on Z2.

**Comment by Pyko1** [comment]
a gib tile, that gibs anyone in it separating the shit. And add more checks to stop things working in z2.

**Comment by ArcLumin** [comment]
Why not just use the tiles we already have, the moving space tiles? Spits them out into another Z level

**Comment by ArcLumin** [comment]
As for using tiles, if you used gibbing tiles or moving space ones, there's still a couple working ways to centcom

**Comment by Pyko1** [comment]
well gotta fix shit that can pass moving space.
